registry.go

```diff
@@ -46,6 +46,9 @@ type Namespace interface {
 
 	// BlobStatter returns a BlobStatter to control
 	BlobStatter() BlobStatter
+
+	// TODO(amidlash): Consider returning notary instead of upstream here.
+	Upstream() string
```

Comment:
I think it is a better to leave this interface alone and just type check for this capability where it is needed.